core/fs/temp.test.ts

```diff
@@ -31,3 +31,47 @@ Deno.test("tempDirectory({ chdir: true }) changes working directory", async () =
   }
   assertEquals(Deno.cwd(), cwd);
 });
+
+Deno.test("tempDirectory({ chdir: true }) works recursively", async () => {
+  const originalCwd = Deno.cwd();
```

Comment:
Simplify the names:

`originalCwd` -> `cwd`
`outerDirectory` -> `outer`
`innerDirectory` -> `inner`